applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py

```diff
@@ -460,17 +497,35 @@ def test_TL_2D_quadrilateral(self):
 
         A,b = self._define_movement(dim)
 
+        self._ResetDisplacementAndPosition(mp)
         self._apply_BCs(bcs,A,b)
-        self._solve(mp)
+        self._solve(mp, builder_and_type, linearize_on_old_iteration)
         self._check_results(mp,A,b)
         self._check_outputs(mp,A,dim)
-
+        self.assertEqual(mp.ProcessInfo[KratosMultiphysics.NL_ITERATION_NUMBER], expected_iterations)
         #self.__post_process(mp)
 
-    def test_TL_3D_tetra(self):
+    def test_TL_2D_quadrilateral_block(self):
+        builder_type = "block_builder"
+        linearize_on_old_iteration = False
+        self._TL_2D_quadrilateral(builder_type, linearize_on_old_iteration,6 )
+
+    def test_TL_2D_quadrilateral_linearized_on_old_iteration(self):
+        builder_type = "block_builder"
+        linearize_on_old_iteration = True
```

Comment:
Shouldn’t it require less iterations in this case?

Reply:
not necessarily....  the linearization on the previous iteration may provide a better or worst starting point. It will be typically better if it is an increment wrt the previous iteration (as happens in real cases) but not if all deformation is applied at once (as it is the case here)

Reply:
well, after the last fix it does. (Apparently always 2 with the new implementation)